File: HW/HW9/TreeNode.java
package HW.HW9;

public class TreeNode {
    int val;

    TreeNode left;

    TreeNode right;

    TreeNode() {}

    TreeNode(int val) { this.val = val; }

    TreeNode(int val, TreeNode left, TreeNode right) {

        this.val = val;

        this.left = left;

        this.right = right;

     }

    public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
        TreeNode node = root;

        while (node != null) { 
            if (p.val > node.val && q.val > node.val) { 
                node = node.right;
            } else if (p.val < node.val && q.val < node.val) { 
                node = node.left;
            } else { 
                return node;
            }
        }
        return node;
    } 
    public static void main(String[] args) {
        
    }

}

File: HW/HW9/TreeNode.py
class TreeNode(object):

    def __init__(self, val=0, left=None, right=None):

        self.val = val

        self.left = left

        self.right = right

    def lowestCommonAncestor(self, root: 'TreeNode', p: 'TreeNode', q: 'TreeNode') -> 'TreeNode':
        node = root

        while node: 
            if p.val > node.val and q.val > node.val: 
                node = node.right
            elif p.val < node.val and q.val < node.val:
                node = node.left
            else:
                return node
    